server/src/main/java/io/druid/query/extraction/LookupExtractionModule.java

```diff
@@ -0,0 +1,209 @@
+  {
+    return ZKPaths.makePath(LookupCoordinatorManager.LOOKUP_LISTEN_ANNOUNCE_KEY, tier);
+  }
+
+  @Override
+  public List<? extends Module> getJacksonModules()
+  {
+    return ImmutableList.of();
+  }
+
+  @Override
+  public void configure(Binder binder)
+  {
+    JsonConfigProvider.bind(binder, ServerModule.ZK_PATHS_PROPERTY_BASE, LookupListeningAnnouncerConfig.class);
+    binder.bind(LookupListeningResource.class);
+    Jerseys.addResource(binder, LookupListeningResource.class);
+    LifecycleModule.register(binder, LookupReferencesManager.class);
+    LifecycleModule.register(binder, LookupResourceListenerAnnouncer.class);
+  }
+}
+
+@Path(ListenerResource.BASE_PATH + "/" + LookupCoordinatorManager.LOOKUP_LISTEN_ANNOUNCE_KEY)
+class LookupListeningResource extends ListenerResource
+{
+  private static final Logger LOG = new Logger(LookupListeningResource.class);
+
+  @Inject
+  public LookupListeningResource(
+      final @Json ObjectMapper jsonMapper,
+      final @Smile ObjectMapper smileMapper,
+      final LookupReferencesManager manager
+  )
+  {
+    super(
+        jsonMapper,
+        smileMapper,
+        new AbstractListenerHandler<LookupExtractorFactory>(new TypeReference<LookupExtractorFactory>()
+        {
+        })
+        {
+          private final Object deleteLock = new Object();
+
+          @Override
+          public synchronized Object post(final Map<String, LookupExtractorFactory> lookups)
+              throws Exception
+          {
+            final Map<String, LookupExtractorFactory> failedUpdates = new HashMap<>();
+            for (final String name : lookups.keySet()) {
+              final LookupExtractorFactory factory = lookups.get(name);
+              try {
+                if (!manager.updateIfNew(name, factory)) {
+                  failedUpdates.put(name, factory);
+                }
+              }
+              catch (ISE ise) {
+                LOG.error(ise, "Error starting [%s]: [%s]", name, factory);
+                failedUpdates.put(name, factory);
+              }
+            }
+            return ImmutableMap.of("status", "accepted", "failedUpdates", failedUpdates);
+          }
+
+          @Override
+          public Object get(String id)
+          {
+            return manager.get(id);
+          }
+
+          @Override
+          public Map<String, LookupExtractorFactory> getAll()
+          {
+            return manager.getAll();
+          }
+
+          @Override
+          public Object delete(String id)
+          {
+            // Prevent races to 404 vs 500 between concurrent delete requests
+            synchronized (deleteLock) {
+              if (manager.get(id) == null) {
+                return null;
+              }
+              if (!manager.remove(id)) {
+                // We don't have more information at this point.
+                throw new RuntimeException(String.format("Could not remove lookup [%s]", id));
```

Comment:
there actually is an RE, will change to something that does auto-formatting. It is caught at a level higher than here and returned as a 500